splink/comparison_level_library.py

```diff
@@ -68,3 +70,58 @@ def create_sql(self, sql_dialect: SplinkDialect) -> str:
 
     def create_label_for_charts(self) -> str:
         return f"Levenshtein distance of {self.col_name} <= {self.distance_threshold}"
+
+
+class DatediffLevel(ComparisonLevelCreator):
+    def __init__(
+        self,
+        col_name: str,
+        date_threshold: int,
+        date_metric: str = "day",  ##TODO: Lock down to sqlglot supported values
+        cast_strings_to_date: bool = False,
+        date_format: str = "%Y-%m-%d",
+    ):
+        """A comparison level using a date difference function
+
+        e.g. abs(date_diff('day', "mydate_l", "mydate_r")) <= 2  (duckdb dialect)
+
+        Args:
+            col_name (str): Input column name
+            date_threshold (int): The threshold for the date difference
+            date_metric (str): The unit of time ('day', 'month', 'year')
+                for the threshold
+            cast_strings_to_date (bool): Whether to cast string columns to date format
+            date_format (str): The format of the date string
+        """
+        super().__init__(col_name)
+        self.date_threshold = date_threshold
+        self.date_metric = date_metric
+        self.cast_strings_to_date = cast_strings_to_date
+        self.date_format = date_format
+
+    def create_sql(self, sql_dialect: SplinkDialect) -> str:
+        sqlglot_dialect_name = sql_dialect.sqlglot_name()
+        date_col = self.input_column(sql_dialect)
+        date_col_l, date_col_r = date_col.names_l_r()
+        col_l_no_dialect = parse_one(date_col_l, read=sqlglot_dialect_name).sql()
```

Comment:
Feels like this could be better, but might need some change to InputColumn

The problem here is that we're relying on the datediff being expressed in sqlglot's base dialect

but then we're interpolating in column names in our specific dialect (e.g. spark, with ``) which sqlglot's base dialect doesn't understand

Perhaps we could manually construct the sqlglot tree here rather than 'parse_one', or perhaps ew could use 'parse_one' with a placeholder name to get the tree, and then populate the column name parts

col_name needs to have a _l and a _r put on it, even if it's an annoying column name like "first name" (with a space)

So you have quoting issues, and if the dialect is spark, you end up with `` quotes,